core/app/models/spree/zone.rb

```diff
@@ -3,6 +3,11 @@ class Zone < Spree::Base
     has_many :zone_members, dependent: :destroy, class_name: "Spree::ZoneMember", inverse_of: :zone
     has_many :tax_rates, dependent: :destroy, inverse_of: :zone
 
+    with_options through: :zone_members, source: :zoneable do
+      has_many :countries, source_type: "Spree::Country"
+      has_many :states, source_type: "Spree::State"
```

Comment:
I'm not very familiar with the Zone code.  How do we know that a `zoneable` will have country and state relationships defined?